### PR TITLE
Use PHPUnit\Framework\TestCase instead of PHPUnit_Framework_TestCase

### PR DESCRIPTION
**Pull request by carusogabriel**
I use the `PHPUnit\Framework\TestCase` notation instead of `PHPUnit_Framework_TestCase` while extending our TestCases. This will help us migrate to PHPUnit 6, that [no longer support snake case class names](https://github.com/sebastianbergmann/phpunit/blob/master/ChangeLog-6.0.md#changed-1).

I just to bumped PHPUnit version to [`4.8.35`](https://github.com/sebastianbergmann/phpunit/blob/master/ChangeLog-4.8.md#4835---2017-02-06), that support this `namespace`.

### CODE REVIEW

File: tests/PHPUnit/Framework/Constraint/HttpResponseText.php [hunk 1]
```diff
@@ -7,7 +7,9 @@
  */
 namespace Piwik\Tests\Framework\Constraint;
 
-class HttpResponseText extends \PHPUnit_Framework_Constraint
+use PHPUnit\Framework\TestCase;
+
+class HttpResponseText extends Constraint
```
Comment (sgiehl) [inline tests/PHPUnit/Framework/Constraint/HttpResponseText.php]: Tests fail here, as `Constraint` class can't be found. Guess you need to adjust the `use` statement

File: tests/PHPUnit/Unit/DeprecatedMethodsTest.php [hunk 2]
```diff
@@ -86,20 +87,21 @@ public function test_deprecations()
         $this->assertDeprecatedMethodIsRemovedInPiwik4('Piwik\Columns\Dimension', 'getType');
     }
```
Comment (sgiehl) [inline tests/PHPUnit/Unit/DeprecatedMethodsTest.php]: could you revert the changes in coding style below?

**Review by sgiehl — CHANGES_REQUESTED** [review]

File: tests/PHPUnit/Framework/Constraint/HttpResponseText.php (hunk 1 above)
Comment (gabriel-caruso) [reply tests/PHPUnit/Framework/Constraint/HttpResponseText.php]: Thank you, I didn't get this one. I'll amend the commit and change it.

File: tests/PHPUnit/Unit/DeprecatedMethodsTest.php (hunk 2 above)
Comment (gabriel-caruso) [reply tests/PHPUnit/Unit/DeprecatedMethodsTest.php]: Sorry, my IDE did. I revert it. Waiting for Travis CI to check the changes I made.

**Review by sgiehl — APPROVED** [review]

**Comment by sgiehl** [comment]
Should be ready to merge. But we also need to update all the tests in our plugin repos...

**Comment by gabriel-caruso** [comment]
@sgiehl Gonna work on that
